src/printer.js

```diff
@@ -3831,7 +3831,7 @@ function nodeStr(node, options) {
   const alternate = preferred === single ? double : single;
 
   let shouldUseAlternateQuote = false;
-  let canChangeDirectiveQuotes = true;
+  let canChangeDirectiveQuotes = !rawContent.includes('\\');
```

Comment:
Before this change, the following was failing:

    prettier tests/quotes/strings.js --debug-check